File: kata/kata-parking-scala/src/main/java/parking/ParkingBuilder.scala
package parking

import scala.collection.immutable.Set
import scala.collection.immutable.HashSet

class ParkingBuilder0 {
  def withSquareSize(size : Int):ParkingBuilder0 = throw new NotImplementedError
  def withPedestrianExit(index: Int):ParkingBuilder0 = throw new NotImplementedError
  def withDisabledBay(index: Int):ParkingBuilder0 = throw new NotImplementedError
  def build(): Parking0 = throw new NotImplementedError
}

class ParkingBuilder {
  private var size: Int = _;
  private var pedestrianIndexes = new HashSet[Int]
  private var disabledIndexes = new HashSet[Int]
  def withSquareSize(size : Int):ParkingBuilder = {
    this.size = size
    this
  }
  def withPedestrianExit(index: Int):ParkingBuilder = {
    pedestrianIndexes += index
    this
  }
  def withDisabledBay(index: Int):ParkingBuilder =  {
    disabledIndexes += index
    this
  }
  def build(): Parking = new Parking(size, pedestrianIndexes, disabledIndexes)
  
}

File: kata/kata-parking-java-solved/src/main/java/victor/kata/parking/ParkingBuilder.java
package victor.kata.parking;

import java.util.HashSet;
import java.util.Set;

/**
 * Builder class to get a parking instance
 */
public class ParkingBuilder {

	private int squareSize;
	private Set<Integer> pedestrianEntries = new HashSet<>();
	private Set<Integer> disabledBays = new HashSet<>();

	public ParkingBuilder withSquareSize(int size) {
		squareSize = size;
		return this;
    }

    public ParkingBuilder withPedestrianExit(int pedestrianExitIndex) {
    	pedestrianEntries.add(pedestrianExitIndex);
    	return this;
    }

    public ParkingBuilder withDisabledBay(int disabledBayIndex) {
    	disabledBays.add(disabledBayIndex);
    	return this;
    }

    public Parking build() {
        return new Parking(squareSize, pedestrianEntries, disabledBays);
    }
}
